devicetypes/smartthings/zigbee-metering-plug.src/zigbee-metering-plug.groovy

```diff
@@ -1,170 +1,194 @@
-                else if (it.value && it.clusterInt == zigbee.SIMPLE_METERING_CLUSTER && it.attrInt == ATTRIBUTE_READING_INFO_SET) {
-                        log.debug "energy"
-                        map.name = "energy"
-                        map.value = zigbee.convertHexToInt(it.value)/getEnergyDiv()
-                        map.unit = "kWh"
-                }
-
-                if (map) {
-                        result << createEvent(map)
-                }
-                log.debug "Parse returned $map"
-        }
-        return result
-    }
-}
-
-def off() {
-    def cmds = zigbee.off()
-    if (device.getDataValue("model") == "HY0105") {
-        cmds += zigbee.command(zigbee.ONOFF_CLUSTER, 0x00, "", [destEndpoint: 0x02])
-    }
-    return cmds
-}
-
-
-def on() {
-    def cmds = zigbee.on()
-    if (device.getDataValue("model") == "HY0105") {
-        cmds += zigbee.command(zigbee.ONOFF_CLUSTER, 0x01, "", [destEndpoint: 0x02])
-    }
-    return cmds
-}
-
-/**
- * PING is used by Device-Watch in attempt to reach the Device
- * */
-def ping() {
-    return refresh()
-}
-
-def refresh() {
-    log.debug "refresh"
-    zigbee.onOffRefresh() +
-    zigbee.electricMeasurementPowerRefresh() +
-    zigbee.readAttribute(zigbee.SIMPLE_METERING_CLUSTER, ATTRIBUTE_READING_INFO_SET)
-}
-
-def configure() {
-    // this device will send instantaneous demand and current summation delivered every 1 minute
-    sendEvent(name: "checkInterval", value: 2 * 60 + 10 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
-    log.debug "Configuring Reporting"
-    return refresh() +
-    	   zigbee.onOffConfig() +
-           zigbee.configureReporting(zigbee.SIMPLE_METERING_CLUSTER, ATTRIBUTE_READING_INFO_SET, DataType.UINT48, 1, 600, 1) +
-           zigbee.electricMeasurementPowerConfig(1, 600, 1) +
-           zigbee.simpleMeteringPowerConfig()
-}
-
-private int getPowerDiv() {
-    isSengledOutlet() ? 10 : 1
-}
-
-private int getEnergyDiv() {
-    isSengledOutlet() ? 10000 : 100
-}
-
-private boolean isSengledOutlet() {
-    device.getDataValue("model") == "E1C-NB7"
+/**
+ *  Copyright 2019 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+import physicalgraph.zigbee.zcl.DataType
+
+metadata {
+    definition (name: "Zigbee Metering Plug", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "oic.d.smartplug", mnmn: "SmartThings",  vid: "generic-switch-power-energy") {
```

Comment:
Now diff for this one is faulty.

Reply:
@dwd-kwon please correct that file so the diff would allow us to continue the code review.